### PR TITLE
Scaling: Consolidate ScalingTab and clean up components

### PR DESCRIPTION
**Pull request by skoeva**
These changes consolidate `ScalingTab` into the `Scaling` module and extract reusable hooks and components to reduce duplication and improve maintainability.

### Summary

- Move `ScalingTab` from `ScalingTab/` into `Scaling/` alongside `ScalingCard`, removing the separate directory
- Extract `useEditDialog` hook from inline `ScalingTab` logic, encapsulating dialog state, form pre-population, and HPA/manual PATCH save logic via `clusterRequest`
- Extract `ScalingEditDialog` component from inline `ScalingTab` JSX
- Extract `MetricTile` local helper in `ScalingMetrics` to reduce repeated `Typography` pairs, and lift CPU/bounds ternaries out of JSX into variables
- Set explicit time ranges per component: 24h/1h-step for `ScalingTab`, 2h/15min-step for `ScalingCard`; make `timeRangeSecs` and step required params in `useChartData`
- Simplify `Alert` usage across components by removing redundant `Box`/`Typography` wrappers
- Add TSDoc field docs to `ScalingMetricsProps`
- Add `useEditDialog` test suite (10 tests) covering dialog state, form pre-population, HPA and manual save paths, `onSaved` callback, and error handling
- Expand `useDeployments` tests with `setSelectedDeployment` coverage and data-reload stability
- Update `useChartData` tests for required params and additional cache/time-range edge cases
- Fix several correctness issues flagged in review: `boundsValue` leak in `ScalingMetrics`, || 0 coercion in HPA inputs, missing guard in `handleSave`, and grid layout wrapping with 5 tiles

Fixes: #100 

### Testing

- [x] Run `cd plugins/aks-desktop && npm test` and ensure the tests pass

### Screenshot

image

### CODE REVIEW

File: plugins/aks-desktop/src/components/Scaling/hooks/useEditDialog.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  /** Error message from the last failed save, or null. */
+  saveError: string | null;
+  /** Opens the dialog, pre-populating form values from the current deployment/HPA state. */
+  handleEditClick: () => void;
+  /** Closes the dialog without saving. */
+  handleClose: () => void;
+  /** Updates the form values. */
+  setEditValues: React.Dispatch<React.SetStateAction<EditValues>>;
+  /** Persists the current form values to Kubernetes via PATCH. */
+  handleSave: () => Promise<void>;
```
[the remark this answers is not in the document]
Comment (skoeva) [reply plugins/aks-desktop/src/components/Scaling/hooks/useEditDialog.ts:48]: not relevant

File: plugins/aks-desktop/package-lock.json [hunk 2]
```diff
@@ -3258,9 +3258,6 @@
         "linux"
       ]
     },
-    "node_modules/@rollup/rollup-linux-arm64": {
```
Comment (illume) [inline plugins/aks-desktop/package-lock.json]: I think this package-lock.json should be removed from the PR.  I think it's accidentally added.

File: plugins/aks-desktop/src/components/Scaling/components/ScalingEditDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            />
+          </>
+        ) : (
+          <TextField
+            label={t('Number of Replicas')}
+            type="number"
+            fullWidth
+            value={editValues.replicas}
+            onChange={e =>
+              onEditValuesChange(v => ({ ...v, replicas: parseInt(e.target.value, 10) || 0 }))
```
[the remark this answers is not in the document]
Comment (skoeva) [reply plugins/aks-desktop/src/components/Scaling/components/ScalingEditDialog.tsx]: done

File: plugins/aks-desktop/src/components/Scaling/ScalingTab.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+          selectedDeployment={selectedDeployment}
+          deployments={deployments}
+          loading={loading}
+          onDeploymentChange={setSelectedDeployment}
+        />
+      </Box>
+
+      {(error || saveError) && (
+        <Alert severity="warning" sx={{ mb: 2 }}>
+          {error ?? saveError}
```
[the remark this answers is not in the document]
Comment (skoeva) [reply plugins/aks-desktop/src/components/Scaling/ScalingTab.tsx:112]: this adds unnecessary complexity
